File: C++/016.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int N, A, B, C;
    cin >> N >> A >> B >> C;
    int ans = 9999;
    for (int i = 0; i < 10000; i++) {
        for (int j = 0; j< 10000; j++) {
            long long k = 1LL * N - A*i - B*j;
            if (k%C == 0 and k >= 0) {
                int candidate = i + j + (k / C);
                if (ans >= candidate) {
                    ans = candidate;
                }
            }
        }
    }
    cout << ans << endl;
}

File: Python/016.py
N = int(input())
A, B, C = map(int, input().split())

ans = 9999

for i in range(10000):
    for j in range(10000):
        k = N - A*i - B*j
        if k % C == 0 and k >= 0:
            candidate = i + j + (k // C)
            if ans > candidate:
                ans = candidate

print(ans)